pkg/platform/runtime/buildexpression/buildexpression.go

```diff
@@ -906,6 +906,31 @@ func (e *BuildExpression) SetDefaultTimestamp() error {
 	return nil
 }
 
+// MaybeSetDefaultTimestamp changes the solve node's "at_time" value to "$at_time" if and only if
+// the current value is the given timestamp.
+// Buildscripts prefer to use variables for at_time and define them outside the buildexpression as
+// the expression's commit time.
+// While modern buildexpressions use variables, older ones bake in the commit time. This function
+// exists primarily to update those older buildexpressions for use in buildscripts.
+func (e *BuildExpression) MaybeSetDefaultTimestamp(ts *strfmt.DateTime) error {
```

Comment:
nit: the prefix maybe got me worried we were introducing some freak functional stuff. We could rename the function or have a function call for a compatibility check followed by `e.setDefaultTimeStamp` in the calling code which I believe is more in line with our codebase. Up to you though.

Reply:
By name it might seem odd, but I believe the doc comment explains nicely what the function is for, and what it does. If there's something not clear, please let me know. Otherwise, please suggest a new name or an example of a compatibility check followed by `e.setDefaultTimeStamp`, because I don't understand what you mean :(

Reply:
I just meant something like

```
be := NewBuildExpression(bytes)
if be.IsIncompatibleTimeStamp(ts) {
   err = be.UpdateTimestamp()
   ...
}
```

But I'm not married to the idea, just a suggestion as it's a pattern I've seen in our codebase before.

Reply:
I see what you mean. I experimented with something like that before. My first inclination was to add a `func (e *BuildExpression) GetTimestamp() (*strfmt.DateTime, error)` method, but since "at_time" can be "$at_time", the function couldn't always return a timestamp, even if the "at_time" field had a valid value.

Then there was the idea of adding `func (e *BuildExpression) TimestampIs(other ts *strfmt.DateTime) (bool, err)`, but it would result in something quite complicated like:

```
if setDefault, err := expr.TimestampIs(other); err == nil && setDefault {
  err := expr.SetDefaultTimestamp()
  if err != nil {
    ...
  }
} else if err != nil {
  ...
}
```

As you can see, a simple

```
err := expr.MaybeSetDefaultTimestamp(other)
if err != nil {
  ...
}
```

seemed better to me :)